packages/x-scheduler/src/primitives/models/event.ts

```diff
@@ -26,19 +26,51 @@ export interface CalendarEvent {
    * The id of the resource this event is associated with.
    */
   resource?: CalendarResourceId;
-
+  /**
+   * The recurrence rule for the event, if it is a recurring event.
+   */
+  rrule?: RRuleSpec;
   /**
    * `true` if the event is an all-day event.
    */
   allDay?: boolean;
 }
 
-export interface CalendarEventWithPosition extends CalendarEvent {
+// RRULE day codes
+export type ByDayCode = 'MO' | 'TU' | 'WE' | 'TH' | 'FR' | 'SA' | 'SU';
+
+// Ordinals for BYDAY: 'MO' | 'WE' | '4SA' | '-1MO' (1..5, negatives for "last")
+type Ordinal = `${'' | '-'}${1 | 2 | 3 | 4 | 5}`;
+export type ByDayValue = ByDayCode | `${Ordinal}${ByDayCode}`;
+
+// Minimal RRULE spec the scheduler needs
+export interface RRuleSpec {
+  freq: 'DAILY' | 'WEEKLY' | 'MONTHLY' | 'YEARLY';
+  interval?: number;
+  /** BYDAY: MO..SU or ordinals like 4SA, -1MO */
+  byDay?: ByDayValue[];
+  /** BYMONTHDAY: 1..31 */
+  byMonthDay?: number[];
+  /** BYMONTH: 1..12 */
+  byMonth?: number[];
+  /** COUNT */
+  count?: number;
+  /** UNTIL (inclusive) */
+  until?: SchedulerValidDate;
+}
+
+export interface CalendarEventOccurence extends CalendarEvent {
+  key: CalendarEventKey;
+}
+
+export interface CalendarEventOccurenceWithPosition extends CalendarEventOccurence {
   eventRowIndex?: number;
 }
 
 export type CalendarEventId = string | number;
 
+export type CalendarEventKey = string | number;
```

Comment:
If this type is only used to define the `key` property, I think we can skip it and hardcode its value.
We have `CalendarEventId` because it's often used in methods and objects, I think the key will not be used a lot except to pass it to the React `key` prop.